server/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java

```diff
@@ -692,7 +689,15 @@ private void syncShardStatsOnNewMaster(List<SnapshotsInProgress.Entry> entries)
                                 snapshot.snapshot(),
                                 shardId,
                                 localShard.getValue().getShardSnapshotResult(),
-                                () -> null
+                                (outcomeInfoString) -> localShard.getValue().updateStatusDescription(
+                                    Strings.format("""
```

Comment:
This could theoretically grow without bound. I figure it's unlikely to happen, since we'd stop once a remote call successfully updates the cluster state. We'd have to keep rolling over the master, and keep failing to get majority, or persistence in the blob store for serverless.

Reply:
Yeah could we impose a limit here just in case? Simply truncating the string if it exceeds 1000 characters or something?

Reply:
Updated 👍 I used substring: a bit wordy, let me know if there's some other/better standard way of doing it.